app/src/androidTest/java/tech/relaycorp/gateway/background/endpoint/GatewaySyncServiceParcelCollectionTest.kt

```
@@ -0,0 +1,107 @@
+
+    @Test
+    fun parcelCollection_receiveParcel() = runBlocking {
+        setGatewayCertificate(CertificationPath.PRIVATE_GW)
+        val parcel = ParcelFactory.buildSerialized()
+        val storeResult = storeParcel.store(parcel, RecipientLocation.LocalEndpoint)
+        assertTrue(storeResult is StoreParcel.Result.Success)
+
+        val parcelCollection =
+            PoWebClient.initLocal(PDCServer.PORT)
+                .collectParcels(
+                    arrayOf(
+                        NonceSigner(
+                            CertificationPath.PRIVATE_ENDPOINT,
+                            KeyPairSet.PRIVATE_ENDPOINT.private
+                        )
+                    ),
+                    StreamingMode.CloseUponCompletion
+                )
+                .first()
+
+        assertEquals(
+            Parcel.deserialize(parcel).id,
+            Parcel.deserialize(parcelCollection.parcelSerialized).id
+        )
+    }
+
+    @Test(expected = ServerConnectionException::class)
+    fun parcelCollection_invalidHandshake() = runBlocking {
+        setGatewayCertificate(CertificationPath.PRIVATE_GW)
+        val parcel = ParcelFactory.buildSerialized()
+        val storeResult = storeParcel.store(parcel, RecipientLocation.LocalEndpoint)
+        assertTrue(storeResult is StoreParcel.Result.Success)
+
+        val parcelCollection =
+            PoWebClient.initLocal(PDCServer.PORT)
+                .collectParcels(
+                    arrayOf(
+                        NonceSigner(
+                            CertificationPath.PRIVATE_ENDPOINT,
+                            KeyPairSet.PUBLIC_GW.private // Invalid key to trigger invalid handshake
+                        )
+                    ),
+                    StreamingMode.CloseUponCompletion
+                )
+                .first()
+
+        assertEquals(
+            Parcel.deserialize(parcel).id,
+            Parcel.deserialize(parcelCollection.parcelSerialized).id
+        )
```

This is never executed, right? I'd suggest to delete it along with the `parcelCollection` variable so it's clear that `PoWebClient.initLocal()...` should throw the exception